### DISCUSSION TITLE
*Language Support for Java(TM) by Red Hat* doesn't work on GitHub Codespaces

### DISCUSSION THREAD

**Discussion by snebjorn** [community/community#35119] [Codespaces]
Language Support for Java(TM) by Red Hat was working fine last week. But now JAVA language support is gone. It doesn't show up in Running Extensions.

Any one else have this problem?
Suggestions?

**Comment by adnamalin** [comment]
👋 Are you using the [default dev container configuration](https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/introduction-to-dev-containers#using-the-default-dev-container-configuration) for your project, that default configuration supports Python, Node, PHP, Java, Go, C++, Ruby, and .NET Core/C#. To support other languages you can set up a [predefined dev container configuration](https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/introduction-to-dev-containers#using-a-predefined-dev-container-configuration) or [custom dev container configuration](https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/introduction-to-dev-containers#creating-a-custom-dev-container-configuration).

**Comment by snebjorn** [comment]
I'm using the mcr.microsoft.com/vscode/devcontainers/java:0-${VARIANT} image.
I can compile the JAVA project just fine from the terminal. So JAVA is installed and working. But JAVA language support in the VSCode editor window is not working. The stuff coming from **Language Support for Java(TM) by Red Hat**.

This is my setup: https://github.com/snebjorn/openapi-generator/tree/feature/devcontainer

**Language Support for Java(TM) by Red Hat** is also suspiciously missing from Running Extensions
![image](https://user-images.githubusercontent.com/1266245/194120869-54a8d031-fa5f-4cf2-b183-afbaee616199.png)

**Comment by adnamalin** [comment]
I've tried creating a codespace in https://github.com/snebjorn/openapi-generator using the browser and VSCode (v1.71.2), I see `Language Support for Java(TM) by Red Hat` installed in both. 

Could you try opening a codespace in the browser and seeing if it's installed in the browser workbench? You can access the `Open in browser` options by going to github.com/codespaces and using the options menu:

![Screen Shot 2022-10-05 at 2 57 17 PM](https://user-images.githubusercontent.com/17835681/194151369-890aee56-8c83-43e9-a511-7f5773393216.png)

**Reply by snebjorn** [reply]
yes it's listed as installed. But is it running? Can you see it in running extensions? 
![image](https://user-images.githubusercontent.com/1266245/194152599-997e6c90-ee54-430f-b77c-b2729cd7ba4d.png)


I can't
![image](https://user-images.githubusercontent.com/1266245/194152296-30ea805f-e735-4f71-aaf2-29613e8f274f.png)

If you open a `.java` file can you code navigate (CTRL+click) in it?

**Reply by adnamalin** [reply]
I don't see `Language Support for Java(TM) by Red Hat` running but I do see `Extension Pack for Java` like you have, and I can code navigate:


https://user-images.githubusercontent.com/17835681/194377643-c75a82ac-b80f-41e4-8b38-13cadac45f71.mov

**Reply by snebjorn** [reply]
very odd

**Reply by adnamalin** [reply]
Could you try opening a codespace in the browser and seeing if it's working in the browser VSCode workbench?

**Reply by snebjorn** [reply]
I tried both in the browser and local VSCode. It's the same.
The screenshot with the running extensions is from the browser

**Reply by adnamalin** [reply]
What version of VSCode are you running locally and in the browser? In the browser you can find this by using the `hamburger menu > help > about`

![Screen Shot 2022-10-06 at 3 02 14 PM](https://user-images.githubusercontent.com/17835681/194407579-25a1d468-c50d-41b4-a88b-4fce5158d15f.png)

Also, in my latest codespace on the repo I do see `Language Support for Java(TM) by Red Hat` running:
![Screen Shot 2022-10-06 at 2 47 24 PM](https://user-images.githubusercontent.com/17835681/194407699-231076b3-eda0-47b5-8760-8acbe39fd0ca.png)

**Reply by snebjorn** [reply]
Running locally (just updated, still no JAVA language support)
```
Version: 1.72.0 (user setup)
Commit: 64bbfbf67ada9953918d72e1df2f4d8e537d340e
Date: 2022-10-04T23:20:39.912Z
Electron: 19.0.17
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Windows_NT x64 10.0.19044
Sandboxed: No
```

Running in browser
```
Version: 1.71.2
Commit: 74b1f979648cc44d385a2286793c226e611f59e7
Date: 2022-09-14T21:04:17.383Z
Browser: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/105.0.0.0 Safari/537.36
```

**Reply by snebjorn** [reply]
What does your Language Support for JAVA output window look like? Mine is completely empty

![image](https://user-images.githubusercontent.com/1266245/194619717-e6723804-436f-4625-8726-27ac1cf52810.png)

**Comment by snebjorn** [comment]
Deleted the codespace machine and made a new instance. Issue persists 😞 

![image](https://user-images.githubusercontent.com/1266245/194642403-264fb7e4-e65c-47f2-8a61-5fa3e019aab5.png)

**Comment by snebjorn** [comment]
when running 
![image](https://user-images.githubusercontent.com/1266245/194650621-94d46bdc-d93a-4499-b9f5-a668232b390d.png)

I got
![image](https://user-images.githubusercontent.com/1266245/194650577-45b40034-b94f-41f2-bcf2-d646f47a5e73.png)

**Comment by testforstephen** [comment]
```
[2022-10-08 12:47:07.919] [window] [error] EPERM: operation not permitted, copyfile '/home/vscode/.vscode-remote/extensions/redhat.java-1.11.0-linux-x64/server/config_ss_linux/config.ini' -> '/home/vscode/.vscode-remote/data/User/globalStorage/redhat.java/1.11.0/config_ss_linux/config.ini': Error: EPERM: operation not permitted, copyfile '/home/vscode/.vscode-remote/extensions/redhat.java-1.11.0-linux-x64/server/config_ss_linux/config.ini' -> '/home/vscode/.vscode-remote/data/User/globalStorage/redhat.java/1.11.0/config_ss_linux/config.ini'
    at Object.copyFileSync (node:fs:2804:3)
    at f (/home/vscode/.vscode-remote/extensions/redhat.java-1.11.0-linux-x64/dist/extension.js:2:718080)
    at /home/vscode/.vscode-remote/extensions/redhat.java-1.11.0-linux-x64/dist/extension.js:2:720919
    at Object.t.prepareExecutable (/home/vscode/.vscode-remote/extensions/redhat.java-1.11.0-linux-x64/dist/extension.js:2:720955)
    at /home/vscode/.vscode-remote/extensions/redhat.java-1.11.0-linux-x64/dist/extension.js:2:698955
    at Generator.next (<anonymous>)
    at /home/vscode/.vscode-remote/extensions/redhat.java-1.11.0-linux-x64/dist/extension.js:2:684461
    at new Promise (<anonymous>)
    at r (/home/vscode/.vscode-remote/extensions/redhat.java-1.11.0-linux-x64/dist/extension.js:2:684206)
    at /home/vscode/.vscode-remote/extensions/redhat.java-1.11.0-linux-x64/dist/extension.js:2:695763
    at new Promise (<anonymous>)
    at /home/vscode/.vscode-remote/extensions/redhat.java-1.11.0-linux-x64/dist/extension.js:2:695747
    at Generator.next (<anonymous>)
    at o (/home/vscode/.vscode-remote/extensions/redhat.java-1.11.0-linux-x64/dist/extension.js:2:684263)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

VS Code Logs show that the user account `vscode` has no permissions within container.

**Comment by snebjorn** [comment]
I can reliably reproduce it using this https://github.com/snebjorn/openapi-generator/tree/37b1132ab917a74249823fe96f3532e85a5881cb/.devcontainer

When changed to https://github.com/snebjorn/openapi-generator/tree/102df74b4cb8d0e8fea14cd3d82ef6d8ff642033/.devcontainer stuff is back to normal
